src/EnergyPlus/BranchInputManager.cc

```diff
@@ -2743,7 +2743,7 @@ namespace BranchInputManager {
         }
 
         // Build node names in branches
-        for (Count = 1; Count <= state.dataBranchInputManager->NumOfBranches; ++Count) {
+        for (Count = 1; Count <= (int)state.dataBranchInputManager->Branch.size(); ++Count) {
             BranchNodes(Count).UniqueNodeNames.allocate(state.dataBranchInputManager->Branch(Count).NumOfComponents * 2);
             BranchNodes(Count).UniqueNodeNames = std::string();
```

Comment:
You just did an `allocate` on this and now you assign it to a blank string because ... ?